src/porting.cpp

```diff
@@ -697,6 +701,26 @@ int mt_snprintf(char *buf, const size_t buf_size, const char *fmt, ...)
 	return c;
 }
 
+bool openURL(std::string url)
+{
+	if (!std::regex_match(url,
+			std::regex(R"(^https?:\/\/([\w\.-]+)\.([a-z\.]{2,6})([\/\w+% \.-]*)*(\?[\w=]*)?\/?$)") )) {
+		errorstream << "Invalid url: " << url << std::endl;
+		return false;
+	}
+
+#if defined(_WIN32)
+	return system((std::string("start '' '") + url + "'").c_str()) == 0;
+#elif defined(__ANDROID__)
+	openURLAndroid(url);
+	return true;
+#elif defined(__APPLE__)
+	return system((std::string("open '") + url + "'").c_str()) == 0;
+#else
+	return system((std::string("xdg-open '") + url + "'").c_str()) == 0;
```

Comment:
nooooooooooooooooooooooooooooooooo

even if this works fine in practice, you should use fork + execvp instead

Reply:
lol, stack overflow lies

How should I get the name of the web browser? Or are you suggesting using fork&execvp on xdg-open instead?

Reply:
yes, use that to call xdg-open

Reply:
actually `posix_spawnp` might be better